packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +130,112 @@ class DeepLinksScreen extends Screen {
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('TODO: build deep link validation tool'),
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage> {
+  String responseBody = 'empty response';
+  int? selectedRowIndex;
+  String searchContent = '';
+  bool bundleByDomain = false;
+
+  void setBundleByDomain(bool shouldBundleByDomain) {
+    setState(() {
+      bundleByDomain = shouldBundleByDomain;
+    });
+  }
+
+  void setResponse(String response) {
+    setState(() {
+      responseBody = response;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Row(
+          children: [
+            Expanded(child: buildTitle('All deep links', context)),
+            InkWell(
+              onTap: () {
+                setBundleByDomain(!bundleByDomain);
+              },
+              child: const Text('Bundle by domain'),
+            ),
+            const SizedBox(width: 10),
+            SearchBar(
```

Comment:
may be too advanced for the current use case, but consider using `SearchField` from `search.dart` in DevTools, which has advanced functionality to hook search into your screen controller and provides some advanced features like navigation through results.

Reply:
SearchableFlatTable seems to highlight the result but I want to filter the data with the search content.